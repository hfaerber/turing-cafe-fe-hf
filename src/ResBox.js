import React from 'react';
import './ResBox.css'
import ResCard from './ResCard'

const ResBox = ({ reservations }) => {
  // map over reservations and create a ResCard for each
  let cards = reservations.map(reservation => {
    return (<ResCard
      key={reservation.id}
      id={reservation.id}
      name={reservation.name}
      date={reservation.date}
      time={reservation.time}
      number={reservation.number}
    />)
  })

  return(
    <div>{cards}</div>
  )
}

export default ResBox


// { id: 18907224, name: 'Christie', date: '8/8', time: '7:00', number: 3 }
