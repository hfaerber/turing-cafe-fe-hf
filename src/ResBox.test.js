import React from 'react'
import { shallow } from 'enzyme';
import ResBox from './ResBox';

describe('ResBox', () => {

  it('should match snapshot with all data passed through correctly', () => {
    const wrapper = shallow(<ResBox
        reservations={[{
          id: 11,
          name: 'Robbie',
          date: '04/11',
          time: '7:45 pm',
          number: 2 }]}
        deleteRes={jest.fn()}
      />)
    expect(wrapper).toMatchSnapshot();
  })
})
