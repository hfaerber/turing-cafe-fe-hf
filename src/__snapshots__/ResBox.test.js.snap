// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResBox should match snapshot with all data passed through correctly 1`] = `
<div
  className="resy-container"
>
  <ResCard
    date="04/11"
    deleteRes={[MockFunction]}
    id={11}
    key="11"
    name="Robbie"
    number={2}
    time="7:45 pm"
  />
</div>
`;
